### PR TITLE
Add underlay proto definition for what's supported in an entity filter.

### PR DESCRIPTION
**Pull request by wchamber**
This will power the frontend to shape the user experience of creating an entity filter.
Move ColumnId and AttributeId next to their respective identified messages.

### CODE REVIEW

File: api/src/main/proto/bio/terra/tanagra/underlay/search.proto [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+package bio.terra.tanagra.underlay;
+option java_package = "bio.terra.tanagra.proto.underlay";
+option java_multiple_files = true;
+
+import "bio/terra/tanagra/underlay/entities.proto";
+import "bio/terra/tanagra/underlay/mapping.proto";
+
+// The schema of how an entity filter can be constructed for a given entity.
+message EntityFiltersSchema {
```
Comment (calbach) [inline api/src/main/proto/bio/terra/tanagra/underlay/search.proto:13]: Same question as on the other PR: are these properties of the filters, or properties of the entity attributes themselves? Is the schema sufficient to express what search operations are possible, or do we need to constraint / expand upon that with this additional structure?

Comment (wchamber) [reply api/src/main/proto/bio/terra/tanagra/underlay/search.proto:13]: I think these are properties of the filters. For example, we might not want to suggest users filter on every attribute on the entity - e.g. we don't need selectors for both `person.gender` and `person.gender_concept_id`, we would just have an EnumHint for `person.gender_concept_id`.
We might need to further expand. Right now the supported operations on the DataType are implicit (not that we've defined many operations yet, just `EQUALS` and `LESS_THAN`). We might want some special string term search operator that's not available on every `STRING` entity, at which point we might want to enumerate the supported attributes, or add sub types for DataTypes. Not sure.

File: api/src/main/proto/bio/terra/tanagra/underlay/search.proto [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+// A hint for a filterable attribute that the values are within certain bounds.
+message IntegerBoundsHint {
+  optional int64 min = 1;
+  optional int64 max = 2;
+}
+
+// A relationship on an entity that may be used in an entity filter.
+message FilterableRelationship {
+  // The name of the relationship.
+  string relationship_name = 1;
```
Comment (calbach) [inline api/src/main/proto/bio/terra/tanagra/underlay/search.proto:76]: Does this need to exactly match the relationship definition in the entity model? If so, should this just act as a lookup into that relationship config? thereby making the `related_entity` here duplicative/unnecessary?

File: api/src/main/proto/bio/terra/tanagra/underlay/search.proto [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  repeated FilterableRelationship relationships = 3;
+}
+
+// An attribute on an entity that may be used for filtering with hints on how to find
+// interesting values.
+message FilterableAttribute {
+  // The attribute that can be used in the entity filter.
+  AttributeId attribute = 1;
+
+  // Hints are usually but not necessarily exclusive.
```
Comment (calbach) [inline api/src/main/proto/bio/terra/tanagra/underlay/search.proto:28]: I'm curious - what is the exception you are imagining here? An attribute which an overloaded purpose?

File: api/src/main/proto/bio/terra/tanagra/underlay/search.proto (hunk 1 above)
Comment (calbach) [reply api/src/main/proto/bio/terra/tanagra/underlay/search.proto:13]: Thanks, I think I'm convinced.

**Review by calbach — APPROVED** [review]

File: api/src/main/proto/bio/terra/tanagra/underlay/search.proto (hunk 3 above)
Comment (wchamber) [reply api/src/main/proto/bio/terra/tanagra/underlay/search.proto:28]: ... I guess I don't have an example. I can make it a oneof for now.

File: api/src/main/proto/bio/terra/tanagra/underlay/search.proto (hunk 2 above)
Comment (wchamber) [reply api/src/main/proto/bio/terra/tanagra/underlay/search.proto:76]: Yep, good catch.

**Review by calbach — APPROVED** [review]

**Pull request merged**